### ISSUE TITLE
--nospaceoperators doesn't work with closure RHS (and maybe LHS)

### ISSUE THREAD

**Issue by jegnux** [nicklockwood/SwiftFormat#843]
In my project I have a an infix `..` operator allowing me to configure any instance on creation:

```swift

let label = UILabel()..{
  $0.textColor = .red
}

// instead of

let label = {
  let label = UILabel()
  label.textColor = .red
  return label
}()
```

Until now, the `spaceAroundOperators` rule formatted it like that:
```swift
let label = UILabel() .. {
  $0.textColor = .red
}
```

But I just discovered the `--nospaceoperators` option, which I tried, but unfortunately it formats the code like that:
```swift
let label = UILabel().. {
  $0.textColor = .red
}
```
Hence, the operator is considered as a _postfix_ operator instead of an _infix_ operator, breaking my code.
I guess it conflicts with the `spaceAroundBraces` rule.

I didn't try with a LHS closure but if my assumption is correct, I guess there's the same issue.

**Comment by nicklockwood** [comment]
@jegnux this looks like a bug, thanks for reporting 👍

**Comment by jegnux** [comment]
@nicklockwood I written a failing test for it:
```swift
    func testRemoveSpaceEvenBeforeRHSClosure() {
        let input = "let label = UILabel() .. { $0.textColor = .red }"
        let output = "let label = UILabel()..{ $0.textColor = .red }"
        let options = FormatOptions(noSpaceOperators: [".."])
        testFormatting(for: input, output, rule: FormatRules.spaceAroundOperators, options: options)
    }
```

It generates 2 errors:
```
XCTAssertEqual failed: ("[:1:1: warning: (spaceAroundBraces) Add or remove space around curly braces.]") is not equal to ("[]")
XCTAssertEqual failed: ("let label = UILabel().. { $0.textColor = .red }") is not equal to ("let label = UILabel()..{ $0.textColor = .red }")
```
I'm not sure to understand the first fail (where is this triggered?)

If I try to add `spaceAroundBraces` in `excluded:` of the `testFormatting`, the test pass, so it seems to confirm my assumption.

I tried to add `spaceAroundBraces` in `orderAfter` of `spaceAroundOperators` but it doesn't fix the issue 😞

**Comment by nicklockwood** [comment]
@jegnux yeah, the issue is that `spaceAroundBraces` doesn't check if the token next to the brace is an operator, so it conflicts with `spaceAroundOperators`. `spaceAroundBraces` should never insert or remove a space if the neighboring token in an operator (except maybe for certain special cases like `.` or `?`) because it's never safe to change whitespace on just one side of an operator as it changes the semantic meaning.

**Comment by jegnux** [comment]
@nicklockwood Thanks for your detailed explanation, it helped me to find a fix! I will open a PR asap ☺️

**Comment by jegnux** [comment]
Fixed in #844 🙂
